istio-telemetry/kiali-operator/templates/clusterrole.yaml

```diff
@@ -0,0 +1,302 @@
+metadata:
+  name: kiali-operator
+  labels:
+    app: kiali-operator
+    release: {{ .Release.Name }}
+rules:
+  - apiGroups: [""]
+    resources:
+    - configmaps
+    - endpoints
+    - events
+    - persistentvolumeclaims
+    - pods
+    - serviceaccounts
+    - services
+    verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups: [""]
+    resources:
+    - namespaces
+    verbs:
+    - get
+    - list
+    - patch
+  - apiGroups: ["apps"]
+    resources:
+    - deployments
+    - replicasets
+    verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups: ["monitoring.coreos.com"]
+    resources:
+    - servicemonitors
+    verbs:
+    - create
+    - get
+  - apiGroups: ["apps"]
+    resourceNames:
+    - kiali-operator
+    resources:
+    - deployments/finalizers
+    verbs:
+    - update
+  - apiGroups: ["kiali.io"]
+    resources:
+    - '*'
+    verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups: ["rbac.authorization.k8s.io"]
+    resources:
+    - clusterrolebindings
+    - clusterroles
+    - rolebindings
+    - roles
+    verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups: ["apiextensions.k8s.io"]
+    resources:
+    - customresourcedefinitions
+    verbs:
+    - get
+    - list
+    - watch
+  - apiGroups: ["extensions"]
+    resources:
+    - ingresses
+    verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups: ["route.openshift.io"]
+    resources:
+    - routes
+    verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups: ["oauth.openshift.io"]
+    resources:
+    - oauthclients
+    verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups: ["monitoring.kiali.io"]
+    resources:
+    - monitoringdashboards
+    verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  # The permissions below are for Kiali itself; operator needs these so it can escalate when creating Kiali's roles
+  - apiGroups: [""]
+    resources:
+    - configmaps
+    - endpoints
+    - namespaces
+    - nodes
+    - pods
+    - pods/log
+    - replicationcontrollers
+    - services
+    verbs:
+    - get
+    - list
+    - watch
+  - apiGroups: ["extensions", "apps"]
+    resources:
+    - deployments
+    - replicasets
+    - statefulsets
+    verbs:
+    - get
+    - list
+    - watch
+  - apiGroups: ["autoscaling"]
+    resources:
+    - horizontalpodautoscalers
+    verbs:
+    - get
+    - list
+    - watch
+  - apiGroups: ["batch"]
+    resources:
+    - cronjobs
+    - jobs
+    verbs:
+    - get
+    - list
+    - watch
+  - apiGroups: ["config.istio.io"]
+    resources:
+    - adapters
```

Comment:
we shouldn't have 37 CRDs in a deployed system.  I have 24.

Reply:
yes, its probably these are legacy from earlier versions of Istio. What I should do here is what the author of the current istio/installer helm chart did - and that is to use `*` for `resources` so we don't have to worry about keeping track. Kiali is meant to be able to at least view (and modify if given the permissions) all the Istio config objects, so I think we can keep it looking like this: https://github.com/istio/installer/blob/master/istio-telemetry/kiali/templates/clusterrole.yaml#L114-L120

```
- apiGroups: 
    - config.istio.io
    - networking.istio.io
    - authentication.istio.io
    - rbac.istio.io
    - security.istio.io
    resources: ["*"]
```

I'll submit a commit for this shortly.

Reply:
done - https://github.com/istio/installer/pull/556/commits/d163a773588f7be077d5d9c8f04b4adb485ef638